TypeCobol/Compiler/Parser/CodeElementBuilder/TypeCobolCodeElementBuilder.cs

```diff
@@ -1,208 +1,421 @@
+                    {
+                        outputs.Add(new CallSiteParameter
+                        {
+                            // TCRFUN_CALL_INOUT_AND_OUTPUT_BY_REFERENCE
+                            SharingMode =
+                                new SyntaxProperty<ParameterSharingMode>(ParameterSharingMode.ByReference, null),
+                            StorageAreaOrValue =
+                                new Variable(CobolExpressionsBuilder.CreateSharedStorageArea(p.sharedStorageArea1())),
+                        });
+                    }
+
+                    int parametersCount = inputs.Count + outputs.Count + inouts.Count;
+                    callSite.Parameters = new CallSiteParameter[parametersCount];
+                    int i = 0;
+
+                    //Add inputs to global callsites parameters
+                    if (inputs.Count > 0)
+                    {
+                        foreach (var param in inputs)
+                        {
+                            callSite.Parameters[i] = param;
+                            i++;
+                        }
+                    }
+
+                    //Add outputs to global callsites parameters
+                    if (outputs.Count > 0)
+                    {
+                        foreach (var param in outputs)
+                        {
+                            callSite.Parameters[i] = param;
+                            i++;
+                        }
+                    }
+
+                    //Add inouts to global callsites parameters
+                    if (inouts.Count > 0)
+                    {
+                        foreach (var param in inouts)
+                        {
+                            callSite.Parameters[i] = param;
+                            i++;
+                        }
+                    }
+
+                    #endregion
+                }
+                else //else it's an error
+                    isInError = true;
+            }
+            else
+            {
+                if (ambiguousSymbol != null &&
+                    (ambiguousSymbol.Type == SymbolType.ProgramEntry || ambiguousSymbol.Type == SymbolType.ProgramName ||
+                     ambiguousSymbol.CandidateTypes.Any(c => c == SymbolType.ProgramEntry || c == SymbolType.ProgramName)))
+                {
+                    //if CandidatesTypes = Program or program Entry
+                    //new ProcedureStyleCallStatement() { ProgramNameOrProgramEntry= temp}
+                    ((AmbiguousSymbolReference) ambiguousSymbolReference.MainSymbolReference).CandidateTypes =
+                       ambiguousSymbol.CandidateTypes.Where(
+                           c => c == SymbolType.ProgramEntry || c == SymbolType.ProgramName).ToArray();
+                    statement =
+                        new ProcedureStyleCallStatement(new ProcedureCall(ambiguousSymbolReference.MainSymbolReference,
```

Comment:
If it's a SymbolReference to ProgramEntry or ProgramName, then there is no ProcedureCall and CallSite